server/util/grpc_forward/grpc_forward.go

```diff
@@ -0,0 +1,74 @@
+package grpc_forward
+
+import (
+	"context"
+	"strings"
+	"sync"
+
+	"github.com/buildbuddy-io/buildbuddy/server/util/flag"
+	"github.com/buildbuddy-io/buildbuddy/server/util/grpc_client"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"github.com/mwitkow/grpc-proxy/proxy"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+var (
+	kytheBackendToProxy = flag.String("app.kythe_backend_to_proxy", "", "If set, proxy kythe traffic to this backend")
+
+	mu                     sync.RWMutex
+	backendConnectionPools map[string]*grpc_client.ClientConnPool
+)
+
+func init() {
+	mu.Lock()
+	backendConnectionPools = make(map[string]*grpc_client.ClientConnPool)
+	mu.Unlock()
+}
+
+type directorFunc func(ctx context.Context, fullMethodName string) (context.Context, *grpc.ClientConn, error)
+
+func getProxyDirector() directorFunc {
+	if *kytheBackendToProxy != "" {
+		return func(ctx context.Context, fullMethodName string) (context.Context, *grpc.ClientConn, error) {
+			target := ""
+			if strings.HasPrefix(fullMethodName, "/kythe.service") {
```

Comment:
If this is meant to be a general forwarding util, there shouldn't be any kythe-specific references either here or in the flag name.

Reply:
Done.